src/main/java/com/voicepin/flow/client/FlowClient.java

```diff
@@ -1,45 +1,179 @@
+import com.voicepin.flow.client.result.GetVoiceprintResult;
+import com.voicepin.flow.client.result.VerifyInitResult;
+import com.voicepin.flow.client.ssl.AnyCertificateStrategy;
+import com.voicepin.flow.client.ssl.CertificateStrategy;
+import com.voicepin.flow.client.ssl.PinnedCertificateStrategy;
+import com.voicepin.flow.client.ssl.TrustedCertificateStrategy;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * @author kodrzywolek
+ * Client for VoicePIN Flow REST API (Text Independent Biometric Voice Verification
+ * server)
+ * 
+ * @author kodrzywolek, Lukasz Warzecha
  */
 public class FlowClient {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(FlowClient.class);
+
     private final Caller caller;
 
-    public FlowClient(String baseURL) {
-        caller = new Caller(baseURL);
+    private FlowClient(FlowClientBuilder builder) {
+        if (builder.username != null) {
+            this.caller = new Caller(builder.baseUrl, builder.username, builder.password, builder.certificateStrategy);
+        } else {
+            this.caller = new Caller(builder.baseUrl);
+        }
     }
 
+    /**
+     * Creates new Voiceprint, which is represented by UUID.
+     * 
+     * @param addVoiceprintRequest
+     * @return assigned voiceprint id
+     * @throws FlowClientException
+     */
     public AddVoiceprintResult addVoiceprint(AddVoiceprintRequest addVoiceprintRequest) throws FlowClientException {
         Call<AddVoiceprintResult> call = new AddVoiceprintCall(addVoiceprintRequest);
         return caller.call(call);
     }
 
+    /**
+     * Gets information about Voiceprint state.
+     * 
+     * @param getVoiceprintRequest voiceprint id
+     * @return voiceprint state
+     * @throws FlowClientException
+     */
+    public GetVoiceprintResult getVoiceprint(GetVoiceprintRequest getVoiceprintRequest) throws FlowClientException {
+        Call<GetVoiceprintResult> call = new GetVoiceprintCall(getVoiceprintRequest);
+        return caller.call(call);
+    }
+
+    /**
+     * Creates new voice model using streamed recording. Sets it for this Voiceprint. The
+     * model will be used as a reference in any subsequent verifications. If the
+     * Voiceprint has already been enrolled, its preexisting model would be overwritten.
+     * 
+     * @param enrollRequest request with voiceprint id and recording stream
+     * @return
+     * @throws FlowClientException
+     */
     public EnrollResult enroll(EnrollRequest enrollRequest) throws FlowClientException {
         Call<EnrollResult> call = new EnrollCall(enrollRequest);
         return caller.call(call);
     }
 
-    public VerifyResult verify(VerifyRequest verifyRequest) throws FlowClientException {
+    /**
+     * Starts verification process on given Voiceprint with incoming stream. Verification
+     * process starts immediately (i.e. it does not wait for the whole stream to be
+     * uploaded).
```

Comment:
Starts verification process on given Voiceprint with provided speech stream. Verification starts immediately (i.e. it does not wait for the whole stream to be sent). The process is finished when the stream ends so in real-time verification it is crucial to listen to verification scores and end the stream when satisfying results are achieved instead of waiting for the final result.